### PR TITLE
feat(geo-layers): Add getTileLoadingState for granular tile loading info

### PR DESCRIPTION
**Pull request by akre54**
## Summary

Adds a new utility function `getTileLoadingState()` that provides detailed information about tile loading progress in TileLayer instances.

## Background

After investigating feedback from PR #10360, I discovered that deck.gl's `layer.isLoaded` behavior is **intentional and correct**:

- `layer.isLoaded` returns `true` once all tile requests are "settled" (completed OR failed)
- This prevents waiting forever for tiles that will never load (404s, network errors, etc.)
- From the source code comments: _"Error / empty tiles resolve to `content === null`. Once Tile2DHeader marks those requests as loaded, do not wait for generated sublayers because there is nothing to render for that tile and `tile.layers` will remain null."_

However, this makes it **impossible to distinguish between "loaded successfully" and "loaded with errors"** using just the boolean `isLoaded` property.

## Solution

`getTileLoadingState()` provides granular information:

```typescript
const state = getTileLoadingState(tileLayer);
// {
//   total: 50,
//   loaded: 47,      // Successfully loaded with content
//   failed: 2,       // 404, network error, etc.
//   pending: 1,      // Still loading
//   percentLoaded: 94,
//   isComplete: false,  // pending > 0
//   isSuccess: false    // failed > 0
// }
```

### 1. Video Export
Wait for all tiles to settle (loaded or failed) before capturing frames:

```typescript
const state = getTileLoadingState(layer);
if (state.isComplete) {
  captureFrame(); // Proceed even if some tiles failed
  if (state.failed > 0) {
    console.warn(`${state.failed} tiles missing in frame`);
  }
}
```

### 2. Progress Indicators
Show detailed loading progress with error counts:

```typescript
<ProgressBar>
  <Progress width={`${state.percentLoaded}%`} />
  <Text>{state.loaded}/{state.total} tiles loaded</Text>
  {state.failed > 0 && <Error>{state.failed} failed</Error>}
</ProgressBar>
```

### 3. Error Recovery
Retry failed tiles up to N times:

```typescript
const state = getTileLoadingState(layer);
if (state.isComplete && !state.isSuccess && retryCount < 3) {
  console.log(`${state.failed} tiles failed, retrying...`);
  layer.setNeedsUpdate(); // Trigger retry
}
```

## Implementation

- Pure utility function (no changes to core layer behavior)
- Checks `tile.content === null` to identify failed tiles
- Fully typed with TypeScript
- Comprehensive tests covering all states
- Complete documentation with examples

## Comparison

**Before (boolean only):**
```typescript
layer.isLoaded  // true - but did tiles load successfully or fail?
```

**After (granular state):**
```typescript
const state = getTileLoadingState(layer);
state.loaded    // 47 tiles succeeded
state.failed    // 2 tiles failed (404, error)
state.pending   // 1 tile still loading
```

## Related

This addresses the feedback from PR #10360 about improving tile loading observability. After investigation, I concluded that:

1. The `isLoaded` behavior is correct by design (prevents deadlocks)
2. A utility function is better than changing core behavior
3. This belongs in `@deck.gl/geo-layers` (as suggested by @chrisgervang)

## Testing

- Unit tests for all state combinations (empty, loading, loaded, failed, mixed)
- Validates that `layer.isLoaded` can be `true` while `state.isSuccess` is `false`
- Tests percentage calculation and edge cases

## Checklist

- [x] Pure utility function (no breaking changes)
- [x] Comprehensive TypeScript types
- [x] Full test coverage
- [x] Documentation with examples
- [x] Exported from `@deck.gl/geo-layers`

### CODE REVIEW

File: modules/geo-layers/src/tile-layer/get-tile-loading-state.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
+// deck.gl
+// SPDX-License-Identifier: MIT
+// Copyright (c) vis.gl contributors
+
+import type {TileLayer} from './tile-layer';
+import type {Tile2DHeader} from '../tileset-2d/tile-2d-header';
+
+export type TileLoadingState = {
```
Comment (chrisgervang) [inline modules/geo-layers/src/tile-layer/get-tile-loading-state.ts]: Please pair this down to the minimum surface. Anything easy to derive doesn't need to be in the api

File: docs/api-reference/geo-layers/get-tile-loading-state.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
+# getTileLoadingState
```
Comment (chrisgervang) [inline docs/api-reference/geo-layers/get-tile-loading-state.md]: Please follow existing documentation patterns.

I'd only expect a basic api reference addition to the base layer getting the feature, and maybe a brief addition to the tips and tricks developer guide of a useful pattern (not all use cases)

File: modules/geo-layers/src/tile-layer/get-tile-loading-state.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ * }
+ * ```
+ *
+ * @note
+ * deck.gl's `layer.isLoaded` returns `true` once all tile requests are "settled"
+ * (completed or failed). This is intentional to prevent waiting forever for tiles
+ * that will never load (404s, network errors, etc.). Use this function to distinguish
+ * between successfully loaded tiles and failed tiles.
+ */
+export function getTileLoadingState<T>(layer: TileLayer<T>): TileLoadingState {
```
Comment (chrisgervang) [inline modules/geo-layers/src/tile-layer/get-tile-loading-state.ts]: Why not a method on TileLayer?
